### DISCUSSION TITLE
OTSF, NDETE and simultaneous events

### DISCUSSION THREAD

**Discussion by mhx-gh** [aesim-tech/simba-project#174] [General]
Looking at https://aesim.tech/doc/ "Predictive Time-Step Solver", it is not clear what mechanism kicks in when an event happens (e.g. a switch is forced to open or close), especially not when multiple events trigger simultaneously. An example of the latter is that the diode in a CCM buck must turn off at the same time the switch closes, especially when the devices are ideal. 

From the diagram it appears that OTSF and NDETE closely cooperate in detecting transitions, and that diode events are maybe purposely excluded from the Newton-Raphson loop (and caught by the interpolator). However, this still wouldn't solve the problem, unless multiple diode events are forbidden or deliberately (randomly?)  rescheduled to not coincide. The latter would impede the quoted relative tolerance? Maybe this is the reason for the amazingly low minimum timestep of 100ps?

**Comment by aesim-tech** [comment]
Hi @mhx-gh  and thanks for your interest in our tool,

* We actually update switch states during the "Linearize Non-Linear Models" block described in the high-level diagram. So the Newton loop converges only when the switch states are _stable_ at a given time step and multiple events can be triggered simultaneously. Have you access to SIMBA? It is very easy to double-check with our design examples (like the CCM buck). There is no rescheduling or whatsoever. 
* Linear interpolation is very important to turn off the diodes at I=0 to avoid unphysical di/dt. 
* The default value for the minimum time step is 1ns and can be changed by the user. This is related to the accuracy of the discontinuity event time and is more important than one would think. In the example you refer to, we use 100ps because we are simulating MOSFET switching transients with time constants in the ns range.
